server/ast/aliased_table_expr.go

```diff
@@ -66,7 +61,7 @@ func nodeAliasedTableExpr(node *tree.AliasedTableExpr) (*vitess.AliasedTableExpr
 	}
 	alias := string(node.As.Alias)
 	if len(alias) == 0 {
-		alias = generateUniqueAlias()
+		alias = utils.GenerateUniqueAlias()
```

Comment:
You don't actually have to do this. The name is AliasedTableExpr, but the alias can be the empty string. Check out how this works in vitess.

Reply:
Here's the relevant rule from the vitess grammar, note that it doesn't set As in many cases:

```yacc
aliased_table_options:
  index_hint_list
  {
    $$ = &AliasedTableExpr{Hints: $1}
  }
| as_opt table_alias index_hint_list
  {
    $$ = &AliasedTableExpr{As: $2, Hints: $3}
  }
| as_of_clause index_hint_list
  {
    $$ = &AliasedTableExpr{AsOf: $1, Hints: $2}
  }
| as_of_clause as_opt table_alias index_hint_list
  {
    $$ = &AliasedTableExpr{AsOf: $1, As: $3, Hints: $4}
  }
```

Reply:
I'm not sure of the context, but GMS fails when some aliases in some children are not set to an alias, even though it's not necessary. I don't know the context, and it's very possible that we technically shouldn't have `AliasedTableExpr` in those contexts where it errors without an alias, but some queries fail without it, so I have to leave it in for now.